cleverhans/attacks.py

```diff
@@ -417,6 +420,10 @@ def generate(self, x, **kwargs):
             eta = adv_x - x
             from cleverhans.utils_tf import clip_eta
             eta = clip_eta(eta, self.ord, self.eps)
+            return i+1, eta
+
+        _, eta = tf.while_loop(
+            cond, body, [tf.zeros([]), eta], back_prop=False)
```

Comment:
Usually our Attack classes make expressions that are differentiable. If you look at the tutorial, you see that for adversarial training we support an option of either back-proping through the attack or not, based on whether we wrap the output of the attack in a stop_gradient op. I'm not very familiar with the `while_loop` implementation. Is there a performance reason to use back_prop=False ahead of time?